### PR TITLE
tests: Update test matrix

### PR DESCRIPTION
**Pull request by liskin**
Python 3.5 is EOL since Sep 2020.
Python 3.9 is out since Oct 2020.
taskwarrior 2.5.2 is out since Dec 2020 (released only on GitHub though).

It's a bit stupid to have to update it manually, but until we figure it
out something better...

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -9,16 +9,16 @@ jobs:
         include:
           - _defaults: # see Dockerfile for variable defaults
           - TASK_VERSION: v2.4.4
-          - TASK_VERSION: 2.5.2
-            continue-on-error: true
+          - TASK_VERSION: v2.5.1
```
Comment (tbabej) [inline .github/workflows/tests.yaml:12]: I guess we want this to be `v2.5.2.`?

Comment (tbabej) [reply .github/workflows/tests.yaml:12]: Actually, scrolled down, realized that's the default now with this PR :)

**Review by tbabej — APPROVED** [review]
Thanks for the PR!

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (liskin) [reply .github/workflows/tests.yaml:12]: Yeah, that's the strategy here: keep reasonable defaults in the Dockerfile for local testing, everything else in the matrix.

**Pull request merged**